### PR TITLE
Add changelog checker for PRs

### PR DESCRIPTION
**Pull request by eldruin**
A bit ironic that adding the checker requires deactivating it, haha

### CODE REVIEW

**Comment by eldruin** [comment]
bors try

**Comment by eldruin** [comment]
Alright, now it works.
bors try

**Comment by eldruin** [comment]
Alright, now it works
bors try

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+on:
+  push:
+    branches: [ staging, trying, master ]
+  pull_request:
```
Comment (Disasm) [inline .github/workflows/changelog.yml:4]: Note that you need this to make it work for PRs from forks: https://github.com/stm32-rs/stm32f4xx-hal/blob/master/.github/workflows/changelog.yml#L1-L2

Comment (eldruin) [reply .github/workflows/changelog.yml]: Ah, I see. Then I can enable the detailed check version instead of the simple one, thanks!

**Review by therealprof — APPROVED** [review]
bors r+

**Comment by eldruin** [comment]
Ah, it seems the changelog checker does not work well with bors since deactivating the changelog check for a PR does not generate any output at all for it in the status and thus bors waits forever for it.
Closing.

**Comment by adamgreig** [comment]
I think you just don't make bors require it, but it can still run for PRs to remind people to add a changelog.

**Comment by eldruin** [comment]
@adamgreig That would be possible, yes. Not great but I guess better than nothing.
Opinions @rust-embedded/hal?

**Review by therealprof — APPROVED** [review]
bors r+

**Pull request merged**